### ISSUE TITLE
GetCredentialsBy {issuer, subject, schema} API calls ignoring query parameters

### ISSUE THREAD

**Issue by frankhinek** [TBD54566975/ssi-service#50]
## Expected Behavior

If a matching credential has been created, based on the following comment:

https://github.com/TBD54566975/ssi-service/blob/03d5868c5b2fcb8297cd586fafac0edb8f7e9cee/pkg/server/router/credential.go#L116

an API call to one of the following:

* `/v1/credentials?issuer=did:key:abc123`
* `/v1/credentials?schema=https://test-schema.com`
* `/v1/credentials?subject=did:key:def456`

should return a JSON array with the matching credential data.

## Current Behavior

If a matching credential has been created, an API call to one of the following:

* `/v1/credentials?issuer=did:key:abc123`
* `/v1/credentials?schema=https://test-schema.com`
* `/v1/credentials?subject=did:key:def456`

returns an error:

```bash
{
    "error": "must use one of the following query parameters: issuer, subject, schema"
}
```

## Steps to Reproduce

1. Create a credential:
    
    ```bash
    curl -X PUT -d '
     {
         "Issuer": "did:abc:123",
         "Subject": "did:def:456",
         "Schema": "https://test-schema.com/name",
         "Data": {
             "firstName": "Jack",
             "lastName": "Dorsey"
         },
         "Expiry": "2022-12-31T05:00:00+00:00"
     }' http://localhost:8080/v1/credentials
    ```
1. Attempt to query by issuer, subject, or schema:
    
    ```bash
    curl -X GET "http://localhost:8080/v1/credentials?issuer=did:abc:123"
    curl -X GET "http://localhost:8080/v1/credentials?subject=did:def:456"
    curl -X GET "http://localhost:8080/v1/credentials?schema=https://test-schema.com/name"
    ```
1. Observe that the following error is returned:
    
    ```bash
    {"error":"must use one of the following query parameters: issuer, subject, schema"}
    ```

## Possible Solution

Assuming that the design intent was for the API user to use a URL query string parameter/value to retrieve credentials, one possible fix is to create a new utility function in [/pkg/server/framework/util.go](https://github.com/TBD54566975/ssi-service/blob/03d5868c5b2fcb8297cd586fafac0edb8f7e9cee/pkg/server/framework/util.go):

```go
// GetQueryValue is a utility to get a parameter value from the query string, nil if not found
func GetQueryValue(r *http.Request, param string) *string {
	value := r.URL.Query().Get(param)
	if value == "" {
		return nil
	}
	return &value
}
```

and modify the `GetCredentials` function:

https://github.com/TBD54566975/ssi-service/blob/03d5868c5b2fcb8297cd586fafac0edb8f7e9cee/pkg/server/router/credential.go#L116-L120

to use `GetQueryValue`:

```go
	issuer := framework.GetQueryValue(r, IssuerParam)
	schema := framework.GetQueryValue(r, SchemaParam)
	subject := framework.GetQueryValue(r, SubjectParam)
```

An alternative could be to use route parameters.  For example:

```
http://localhost:8080/v1/credentials/issuer/did:abc:123
```

I tested this as well, but found that if the intent is to support Schema IDs like `https://test-schema.com/name` (which appears in the current tests), there are issues with the routes being interpreted properly by [httptreemux](https://github.com/dimfeld/httptreemux).

## Environment

```
macOS 12.4
go version go1.18.3 darwin/arm64
Mage Build Tool 1.13.0
```

**Comment by decentralgabe** [comment]
Thank you for the thorough report @frankhinek I'll make sure to get this fixed.

**Comment by decentralgabe** [comment]
Fixed in #54
